src/internal/sdata/sql.go

```diff
@@ -2,36 +2,82 @@ package sdata
 
 import (
 	"fmt"
+	"reflect"
 	"strings"
 
 	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
 	"github.com/pachyderm/pachyderm/v2/src/internal/pachsql"
 )
 
+// limit the number of rows
+const rowLimit = 1000
+
 // SQLTupleWriter writes tuples to a SQL database.
 type SQLTupleWriter struct {
 	tx              *pachsql.Tx
+	tableInfo       *pachsql.TableInfo
 	insertStatement string
+	buf             []interface{}
```

Comment:
Making this `[]Tuple` so you don't have to keep checking for the column length seems good, unless then executing the actual query is annoying. Speaking of which, do we validate anywhere that the tuple lengths are compatible? seems like it would be good to error early